### PR TITLE
Liveness and Readiness Probes

### PR DESCRIPTION
**Pull request by hope-yu-fivestars**

### CODE REVIEW

File: docs/probes.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Liveness and Readiness Probes
+
+Liveness and readiness probes are a way for Kubernetes to constantly monitor your application and performing restarts or holding off traffic as needed. From the [Kubernetes documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/):
```
Comment (agianey) [inline docs/probes.md]: I may be wrong but I think 
"and performing restarts or holding off traffic as needed" should be 
"by performing restarts or holding off traffic as needed" or
"and perform restarts or hold off traffic as needed"

File: docs/probes.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+In the `aladdin-demo` container in [aladdin-demo.deploy.yaml](../helm/aladdin-demo/templates/aladdin-demo.deploy.yaml), we make an http request to the `ping` endpoint of the aladdin-demo app.
+```yaml
+# Readiness probe stops traffic to this pod if it is not ready, wait until it is ready
+readinessProbe:
+  httpGet:
+    path: /ping
+    port: {{ .Values.app.port }}
+  initialDelaySeconds: {{ .Values.app.readiness.initalDelay }}
+  periodSeconds: {{ .Values.app.readiness.period }}
+# Liveness probe terminates and restarts the pod is unhealthy
```
Comment (agianey) [inline docs/probes.md]: should be "if unhealthy"?

**Review by agianey — APPROVED** [review]
Feel free to merge after addressing minor comments

File: helm/aladdin-demo/values.yaml [hunk 3]
```diff
@@ -5,6 +5,14 @@ app:
...
+    use: true
+    initialDelay: 15
+    period: 120
+  readiness:
+    use: true
+    initialDelay: 5
+    period: 60
 
 deploy:
   # number of seconds for the containers to perform a graceful shutdown, after which it is voilently terminated
```
Comment (vinayagg) [inline helm/aladdin-demo/values.yaml:18]: which value are we talking about? 
It may be more obvious if we place the comment on same line as the value.

**Pull request merged**